src/engine/sim_engine/pool.rs

```diff
@@ -925,6 +929,30 @@ impl Pool for SimPool {
         }
     }
 
+    fn encrypt_pool(
+        &mut self,
+        _: &Name,
+        _: PoolUuid,
+        enc: &InputEncryptionInfo,
+    ) -> StratisResult<CreateAction<EncryptedDevice>> {
+        self.encryption_info = convert_encryption_info(Some(enc), None)?;
+        Ok(CreateAction::Created(EncryptedDevice))
+    }
+
+    fn reencrypt_pool(&mut self, _: &Name) -> StratisResult<ReencryptedDevice> {
```

Comment:
Please change sim engine's reencrypt so that it returns an error if pool is not encrypted. Better for stratis-cli testing.